### ISSUE TITLE
Do not work well with TypeScript project references

### ISSUE THREAD

**Issue by JounQin** [vuejs/language-tools#3638]
```jsonc
// tsconfig.json
{
  "files": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    }
  ]
}
```

run `yarn vue-tsc --noEmit`, no error

reproduction: https://github.com/JounQin/test/tree/repro/vue-tsc_yarn3

step: `yarn install && yarn lint`

`src/Test.vue` should error like

```log
yarn vue-tsc --noEmit src/Test.vue
src/Test.vue:2:8 - error TS2339: Property 'web3' does not exist on type 'Window & typeof globalThis'.

2 window.web3

Found 1 error in src/Test.vue:2
```

**Comment by ferm10n** [comment]
I believe you need a `--build` flag in order to use project references

**Comment by ferm10n** [comment]
however I'm finding a different issue, because project references forbid disabling `noEmit`, but vue-tsc requires it

```
❯ yarn vue-tsc --build --watch --noEmit
error TS5094: Compiler option '--noEmit' may not be used with '--build'.
```

```
❯ yarn vue-tsc --build --watch
[6:21:26 p.m.] Starting compilation in watch mode...

js emit is not supported
```

it doesn't matter that the root tsconfig sets `noEmit: false`

**Comment by rstaake-sd** [comment]
We use the command `vue-tsc --build --emitDeclarationOnly tsconfig.file1.json tsconfig.file2.json tsconfig.file3.json` to test multiple files together that contain project references and also reference each other.

As this command generates the needed output to resolve references, I would expect that this would work.

The default `tsc` command is able to handle this setup, but it breaks if vue files are involved and `vue-tsc` is needed.

**Comment by lancatlin** [comment]
This issue seems to be solved in 2.0.10 now.

**Comment by rstaake-sd** [comment]
True. I checked my setup and the one @JounQin described and I did not notice any issue anymore.
